wiki/mapping/intermediate-mapping.md

```diff
@@ -4,15 +4,201 @@ prev: ./basic-mapping.md
+
+### Note Jump Speed & Spawn Distance
+Note Jump Speed (NJS), Spawn Offset (or offset), and Spawn Distance are additional factors you need to consider for visibility. 
+
+Slower NJS is useful for visibility on easier songs where players need to see things coming to react. Faster NJS is useful for fast songs which would otherwise appear cluttered without note spacing. Faster NJS also helps for visually cluttered maps which heavily use elements such as bombs or sliders that otherwise create visual noise. 
+
+Larger spawn distances are good for maps which need high visibility to catch unusual or large motions such as dancey maps. Shorter spawn distances are useful for maps which have simpler patterns but are faster- a shorter spawn distance means that fewer notes are on screen, therefore being less cluttered. 
+
+Offset is used to adjust a song’s spawn distance while keeping the NJS the same, since spawn distance varies with NJS and BPM. Spawn offsets can be positive or negative and usually don't go above 1 or below -1. Below is a table of NJS, spawn offset, and spawn distance values that can be used as a guide (but not as law) for determining your map’s parameters. Using this table, adjust the offset as needed to reach the target spawn distance.
+
+| **Speed & Style** | **Spawn Distance Target** | **Expert NJS** | **Expert+ NJS** |
+|:----:|:----:|:----:|:----:|
+| Slow (< 100 BPM) Standard | 24-27 | 14-16 | 18 |
+| Slow (< 100 BPM) Dancey | 25-28 | 13-15 | 16-18 |
+| Average (100 - 160 BPM) Standard | 24-27 | 15-16 | 17-20 |
+| Average (100 - 160 BPM) Dancey | 25-28 | 14-17 | 16-18 |
+| Fast (> 160 BPM) Standard | 23-27 | 16-17 | 17-20 |
+| Ultrafast (> 200 BPM) Standard | 23-25 | 18-21 | 19-24 |
+
+## Understanding Flow
+In [Basic Mapping](../mapping/basic-mapping.md#do-mapping-with-flow) we covered the bare essentials of parity. Now we’ll dig deeper and explore why some hits just feel bad.  
+
+Flow is arguably the most important concept in mapping - how each arm travels to move from one hit to the next. This can be as simple as alternating between “forehand” and “backhand” notes, or as complicated as thinking about how one arm interacts with the other at the edges of the playspace. Since the player must see blocks in order to react to them, flow highly depends on speed and difficulty. Slower songs are far more forgiving and flexible with regards to flow, while a fast song is highly limited.
+
+### Block Directions & Positions
+Each swing a player makes leaves their arm in a new position. At hard difficulty and higher, the player will leave their arms in this new position while waiting for the next motion. The block that follows must consider the position of the arm.
+
+| **Poor Positioning** | **Poor Angles** | **Good Position & Angle** |
+|:----:|:----:|:----:|
+| *Picture Here* | *Picture Here* | *Picture Here* |
```

Comment:
This is fine, but just wanted to check that these placeholders were intentional.